rust/stub/src/pe_loader.rs

```diff
@@ -179,12 +178,10 @@ impl Image {
         let status = (self.entry)(handle, unsafe { system_table.unsafe_clone() });
 
         // If the kernel entry point returned, deallocate its image, and restore our loaded image handle.
-        // If it calls Exit(), then we don't get here, and our resources are leaked.
-        // I don't see any way to avoid this.
+        // If the kernel exits boot services, we don't get here and there is no point in freeing memory.
         system_table
             .boot_services()
-            .free_pages(self.image.as_ptr() as u64, bytes_to_pages(self.image.len()))
-            .unwrap_or(());
+            .free_pages(self.image.as_ptr() as u64, bytes_to_pages(self.image.len()))?;
```

Comment:
This one is interesting. My thought was "failing to deallocate memory is just a resource leak". However, if this fails (and the UEFI is not extremely buggy), we have attempted a double free because the kernel already freed memory it shouldn't have freed. Given that this violates the `unsafe` contract, I'm actually tempted to panic here instead.

Reply:
Fair enough. In this case, just leave a comment to explain why.